core/src/main/java/org/eclipse/hono/util/ConnectionDuration.java

```diff
@@ -0,0 +1,88 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.util;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.eclipse.hono.annotation.HonoTimestamp;
+
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The resource limits definition corresponding to the connection duration.
+ */
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+public class ConnectionDuration {
+
+    @JsonProperty(value = TenantConstants.FIELD_EFFECTIVE_SINCE, required = true)
+    @HonoTimestamp
+    private Instant effectiveSince;
+
+    @JsonProperty(TenantConstants.FIELD_MAX_MINUTES)
+    private long maxMinutes = TenantConstants.UNLIMITED_MINUTES;
+
+    @JsonProperty(TenantConstants.FIELD_PERIOD)
+    private ResourceLimitsPeriod period;
+
+    /**
+     * Gets the point in time on which the connection duration limit came into effect.
+     *
+     * @return The instant on which the connection duration limit came into effective or 
+     *         {@code null} if not set.
+     */
+    public final Instant getEffectiveSince() {
+        return effectiveSince;
+    }
+
+    /**
+     * Sets the point in time on which the connection duration limit came into effect.
+     *
+     * @param effectiveSince the point in time on which the connection duration limit came into effect
+     *                       and it comply to the {@link DateTimeFormatter#ISO_OFFSET_DATE_TIME}.
+     * @return  a reference to this for fluent use.
+     */
+    public final ConnectionDuration setEffectiveSince(final Instant effectiveSince) {
+        this.effectiveSince = effectiveSince;
+        return this;
+    }
+
+    /**
+     * Gets the maximum device connection duration in minutes to be allowed for the time period 
+     * defined by the {@link TenantConstants#FIELD_PERIOD_MODE} and 
+     * {@link TenantConstants#FIELD_PERIOD_NO_OF_DAYS}.
+     *
+     * @return The maximum number of minutes or {@link TenantConstants#UNLIMITED_MINUTES}
+     *         if not set.
+     */
+    public final long getMaxMinutes(){
+        return maxMinutes;
+    }
+
+    /**
+     * Sets the maximum device connection duration in minutes to be allowed for the time period
+     * defined by the {@link TenantConstants#FIELD_PERIOD_MODE} and 
+     * {@link TenantConstants#FIELD_PERIOD_NO_OF_DAYS}.
+     *
+     * @param maxMinutes The maximum connection duration in minutes to be allowed.
+     * @return  a reference to this for fluent use.
+     * @throws IllegalArgumentException if the maximum number of minutes is set to less than -1.
+     */
+    public final ConnectionDuration setMaxMinutes(final long maxMinutes) {
```

Comment:
I would rather have the name indicate the purpose than the unit of the argument, e.g. *setMaxDuration*. The parameter name and JavaDoc already indicate the unit, don't they?

Reply:
I set it to _setMaxMinutes_ to keep in alignment with the existing naming convention in _resource-limits_ namely`DataVolume.setMaxBytes(...)`. _setMaxDuration_ makes more sense and we could also rename later `DataVolume.setMaxBytes(...)` to `DataVolume.setMaxVolume(...)`.

Reply:
Hm, now that you mention the `DataVolume` spec ... The whole class is already named `ConnectionDuration` so it probably is obvious that we are setting the max duration. Let's stick with *maxMinutes* and *maxBytes*. At least, this way it is consistent and we do not need to rename existing properties.